File: models/category.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Category represents a product category (e.g., "Men's Shirts", "Women's Dresses")
type Category struct {
	ID          primitive.ObjectID `bson:"_id,omitempty"        json:"id,omitempty"`
	Name        string             `bson:"name"                 json:"name"`
	Slug        string             `bson:"slug"                 json:"slug"`                // URL-friendly (e.g., "mens-shirts")
	ParentID    primitive.ObjectID `bson:"parent_id,omitempty"  json:"parent_id,omitempty"` // Optional: For hierarchical categories
	Description string             `bson:"description"          json:"description"`
	Image       string             `bson:"image"                json:"image"`     // URL to category image
	Avatar      string             `bson:"avatar,omitempty"     json:"avatar,omitempty"` // Path to a flat icon under /uploads/avatars/categories/ (e.g. "/uploads/avatars/categories/shirt.svg")
	IsActive    bool               `bson:"is_active"            json:"is_active"`           // Whether the category is active (defaults to true)
	ShowInHeader bool              `bson:"show_in_header"       json:"show_in_header,omitempty"`
	CreatedAt   time.Time          `bson:"created_at,omitempty" json:"created_at,omitempty"`
	UpdatedAt   time.Time          `bson:"updated_at,omitempty" json:"updated_at,omitempty"`
}

// Note: This model requires the following indexes:
// - slug (Unique)
// - parent_id (For hierarchical queries)

File: front_end/src/types/category.ts
export interface Category {
  id?: string;                  // MongoDB ObjectID, can be absent if not yet created
  name: string;
  slug: string;
  parent_id?: string | null;    // Parent category ObjectID as string, nullable
  description: string;
  image: string;                // URL to category image
  avatar?: string;              // Path to a flat icon under /uploads/avatars/categories/ (e.g. "/uploads/avatars/categories/shirt.svg")
  created_at?: string;          // ISO 8601 timestamp, optional
  updated_at?: string;          // ISO 8601 timestamp, optional
  is_active?: boolean;          // Whether the category is active
  show_in_header?: boolean;    // Should this category appear in the site header navigation?
}

export interface CategoryAvatar {
  name: string;   // base name without color suffix and extension (e.g. "shirt")
  color: string;  // "white" or "blue"
  file: string;   // file name on disk (e.g. "shirt-white.svg")
  path: string;   // public URL path (e.g. "/uploads/avatars/categories/shirt-white.svg")
  size: number;   // file size in bytes
}
